src/zenml/cli/artifact.py

```diff
@@ -185,38 +219,44 @@ def delete_artifact(
 def prune_artifacts(
     only_artifact: bool = False, only_metadata: bool = False, yes: bool = False
 ) -> None:
-    """Delete all unused artifacts.
+    """Delete all unused artifacts and artifact versions.
 
     Args:
-        only_artifact: If set, only delete the artifact but not its metadata.
-        only_metadata: If set, only delete metadata and not the actual artifact.
+        only_artifact: If set, only delete the physical artifacts but not the
+            metadata.
+        only_metadata: If set, only delete metadata and not the physical
+            artifacts.
         yes: If set, don't ask for confirmation.
     """
     client = Client()
-    unused_artifacts = depaginate(
-        partial(client.list_artifacts, only_unused=True)
+    unused_artifact_versions = depaginate(
+        partial(client.list_artifact_versions, only_unused=True)
     )
 
-    if not unused_artifacts:
-        cli_utils.declare("No unused artifacts found.")
+    if not unused_artifact_versions:
+        cli_utils.declare("No unused artifact versions found.")
         return
 
     if not yes:
         confirmation = cli_utils.confirmation(
-            f"Found {len(unused_artifacts)} unused artifacts. Do you want to "
-            f"delete them?"
+            f"Found {len(unused_artifact_versions)} unused artifact versions. "
+            f"Do you want to delete them?"
         )
         if not confirmation:
             cli_utils.declare("Artifact deletion canceled.")
             return
 
-    for unused_artifact in unused_artifacts:
+    for unused_artifact_version in unused_artifact_versions:
         try:
-            Client().delete_artifact(
-                name_id_or_prefix=unused_artifact.id,
+            Client().delete_artifact_version(
+                name_id_or_prefix=unused_artifact_version.id,
                 delete_metadata=not only_artifact,
                 delete_from_artifact_store=not only_metadata,
             )
+            unused_artifact = unused_artifact_version.artifact
+            if not unused_artifact.versions and not only_artifact:
+                Client().delete_artifact(unused_artifact.id)
+
         except Exception as e:
             cli_utils.error(str(e))
-    cli_utils.declare("All unused artifacts deleted.")
+    cli_utils.declare("All unused artifacts and artifact versions deleted.")
```

Comment:
Nice little add-on would maybe be to track the space saved? (For future team)

Reply:
TBH I'd say not worth the effort - not many people will ever use these commands anyways I expect 😅